lib/cc/analyzer/engine.rb

```diff
@@ -33,7 +33,7 @@ def run(stdout_io, container_listener)
 
         container.on_output("\0") do |output|
           unless output_filter.filter?(output)
-            stdout_io.write(output)
+            container.stop unless stdout_io.write(output)
```

Comment:
this trailing conditional seems odd to me because `sdtout_io.write` is the primary action here. What do you think of

```ruby
stdout_io.write(output) or container.stop
```

I don't usually advocate using `or` but given that it's for control flow and not logic, I think it's a good use case.